### PR TITLE
Revise PlantLoopSolver rules for series active branches with no flow request

### PR DESCRIPTION
**Pull request by EnergyArchmage**
This pull request revises how plant flows get redistributed so that series active branches only receive flow if they had a non-zero total branch flow request.  Series active branches are those with multiple active components on a single branch.  Parallel chiller plants with branch pumps associated with each chiller are a common example.  The change here avoids sending excess flow down a branch that did not initially have any flow request.  This helps branches stay "off" that did not want to be "on."

Address issue #7400 and maybe #3522

Pull request overview
---------------------
Please change this line to a description of the pull request, with useful supporting information including whether it is a new feature, or fixes a defect, a cross reference to any defects addressed in this PR, the type of changes to be made, and whether diffs are expected/justified based on this change.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by Myoldmopar** [comment]
I'll review this next, probably tomorrow.

File: src/EnergyPlus/Plant/PlantLoopSolver.cc [hunk 1]
```diff
@@ -1812,32 +1812,21 @@ namespace EnergyPlus {
...
+                    // This comment above is not true, for series active branches, DetermineBranchFlowRequest does scan down the branch's
+                    // components already, no need to loop over components
                     BranchMinAvail = Node(LastNodeOnBranch).MassFlowRateMinAvail;
                     BranchMaxAvail = Node(LastNodeOnBranch).MassFlowRateMaxAvail;
                     //            !sum the branch flow requests to a total parallel branch flow request
-                    if (this_splitter_outlet_branch.ControlType == ControlType_Active ||
-                        this_splitter_outlet_branch.ControlType == ControlType_SeriesActive) {
+                    bool activeBranch = this_splitter_outlet_branch.ControlType == ControlType_Active;
+                    bool isSeriesActiveAndRequesting = (this_splitter_outlet_branch.ControlType == ControlType_SeriesActive) && (BranchFlowReq > 0.0);
+                    if (activeBranch || isSeriesActiveAndRequesting ) { // revised logic for series active
```
Comment (Myoldmopar) [inline src/EnergyPlus/Plant/PlantLoopSolver.cc:1829]: @EnergyArchmage I just pushed a commit that makes no functional change, but makes the code (and the GitHub diff) easier to read.  I created a couple temporary variables for each of these IF blocks to make it more clear of the intent of the IF block.  All those parentheses we had in there were masking behavior.

File: src/EnergyPlus/Plant/PlantLoopSolver.cc [hunk 2]
```diff
@@ -1812,32 +1812,21 @@ namespace EnergyPlus {
...
-                        if ((this_comp.TypeOf_Num != TypeOf_PumpVariableSpeed) &&
-                            (this_comp.TypeOf_Num != TypeOf_PumpBankVariableSpeed)) {
-                            continue;
-                        }
-
-                        CompInletNode = this_comp.NodeNumIn;
-                        BranchFlowReq = max(BranchFlowReq, Node(CompInletNode).MassFlowRateRequest);
-                    }
-
+                    // This comment above is not true, for series active branches, DetermineBranchFlowRequest does scan down the branch's
```
Comment (Myoldmopar) [inline src/EnergyPlus/Plant/PlantLoopSolver.cc:1822]: OK, that makes sense.  I cleaned out the block you had commented.

**Review by Myoldmopar — APPROVED** [review]
I have no problem with these changes.  I'm going to run it with a couple defect files and we'll see if this can drop in tomorrow.

**Comment by Myoldmopar** [comment]
@EnergyArchmage, I took a pass at just refining your changes, and once @rraustad caught my little typo, I think I have it back to where you had it.  There are still three files with diffs, which is what I expect.  Can you briefly just confirm that the code changes are what you intended?

Thanks!

**Comment by EnergyArchmage** [comment]
Yeah I reviewed your changes and looks good.  
At some point someone should circle back to the ancient defect file and see if the old issue is fixed.

**Comment by Myoldmopar** [comment]
Code changes are good.  Built locally and ran diffs, they are indeed tiny.  I'm now verifying the original defect file(s).

**Comment by Myoldmopar** [comment]
I went all the way back to the original 3522 defect file and transitioned it up to the latest version.

Both branch chillers are on during the day, but the second chiller isn't on until a little later than the first one:
image

The mass flow rate is following this pattern nicely, with the second branch pump staying off during that delay until the chiller is on:
image

Does that look like what you'd expect @EnergyArchmage?

**Comment by Myoldmopar** [comment]
It kinda looks like similar behavior is exhibited in the develop branch.  Unfortunately there are a lot of defect files related to 3522.  I'm inclined to go ahead and merge this in as it is a clear improvement of the flow request logic.  CI is partially down so I don't think I'm going to click merge right now, but unless someone thinks otherwise I'm going to drop this in once CI wraps.

**Pull request merged**